airbyte-integrations/connectors/source-db2/src/main/java/io.airbyte.integrations.source.db2/Db2Source.java

```diff
@@ -118,8 +118,46 @@ protected String getCountColumnName() {
   }
 
   private CheckedFunction<Connection, PreparedStatement, SQLException> getPrivileges() {
+    // Get user, group and role SELECT privileges from the information schema
```

Comment:
Thanks for the contribution!

Do you think it is possible to use JDBC's function `getTablePrivileges` for this checking? This would be more generic, and could be used by others too. For instance:

Here is an example implementation where I check against `SELECT` for the current user:


```
  @Suppress("UNCHECKED_CAST")
    @Throws(SQLException::class)
    override fun <T> getPrivilegesTableForCurrentUser(
        database: JdbcDatabase?,
        schema: String?
    ): Set<T> {
        val rs = database!!.metaData.getTablePrivileges(null, schema, "%")
        val result = mutableSetOf<T>()
        while (rs.next()) {
            val grantee = rs.getString("GRANTEE")
            val privilege = rs.getString("PRIVILEGE")
            if (grantee == database.metaData.userName && privilege == "SELECT") {
                val tableName = rs.getString("TABLE_NAME")
                val schemaName = rs.getString("TABLE_SCHEM")
                result.add(JdbcPrivilegeDto(grantee, tableName, schemaName, privilege) as T)
            }
        }
        return result.toSet()
    }
```

Reply:
Hi @theyueli, I guess it could work but I'm not a JDBC expert.